### PR TITLE
COM-662 Impelement indicator 2b, 2c and 2d

### PR DESCRIPTION
**Pull request by lauravignoli**

### CODE REVIEW

File: reportssql/treatment_report_functions.sql [hunk 1]
```diff
@@ -431,3 +434,37 @@ BEGIN
...
+    JOIN concept c ON do.duration_units = c.concept_id AND c.retired = 0
+    JOIN drug d ON d.drug_id = do.drug_inventory_id AND d.retired = 0
+    WHERE o.patient_id = p_patientId AND o.voided = 0
+        AND drugIsARV(d.concept_id)
+        AND calculateTreatmentEndDate(
+            o.scheduled_date,
+            do.duration,
+            c.uuid -- uuid of the duration unit concept
+            ) BETWEEN timestampadd(MONTH, p_drugDurationMonthsMin, o.scheduled_date) 
+                AND timestampadd(MONTH, p_drugDurationMonthsMax, o.scheduled_date)
```
Comment (tmvumbi2) [inline reportssql/treatment_report_functions.sql]: @lauravignoli the current implementation will not work well when the duration is a decimal.
If for example, we need to check that the duration is between 0 and 2 months; and there is a drug ordered on 10 Jan with a duration of 2 months and a half; the math will be:
- treatmentEndDate = 25 Mar (10 Jan + 2 months and 15 days)
- interval1 = 10 Jan (10 Jan + 0 month)
- interval2 = 10 Mar (10 Jan + 2 months)

25 Mar  BETWEEN 10 Jan AND 10 Mar => False

A possible fix is to create a function that calculates 'do.duration' to months (that discards the decimal, i.e. 2.8 months should return 2 months); and the condition could be something like:
``` getDurationInMonths(do.duration, c.uuid) BETWEEN p_drugDurationMonthsMin AND p_drugDurationMonthsMax```

File: reportssql/treatment_report_functions.sql [hunk 2]
```diff
@@ -431,3 +434,37 @@ BEGIN
...
+DECLARE result TINYINT(1) DEFAULT 0;
+
+    SELECT true INTO result
+    FROM orders o
+    JOIN drug_order do ON do.order_id = o.order_id
+    JOIN concept c ON do.duration_units = c.concept_id AND c.retired = 0
+    JOIN drug d ON d.drug_id = do.drug_inventory_id AND d.retired = 0
+    WHERE o.patient_id = p_patientId AND o.voided = 0
+        AND drugIsARV(d.concept_id)
+        AND calculateTreatmentEndDate(
```
Comment (tmvumbi2) [inline reportssql/treatment_report_functions.sql:464]: @lauravignoli the duration needs to be checked on ARV satisfying the following conditions:
(1) if the current patient is an old patient: an ARV that was dispensed within the reporting period OR was dispensed before the reporting period but with a treatment that covers the entire reporting period
(2) if the current patient is not an old patient: an ARV that was dispensed within the reporting period

The function `isOldPatient(p_patient_id, p_startDate)` checks if a patient is old or not

File: reportssql/treatment_report_functions.sql (hunk 1 above)
Comment (lauravignoli) [reply reportssql/treatment_report_functions.sql]: @tmvumbi2 Why would we use the duration with decimals? If you want we can add a comment that the months must be integer values?

File: reportssql/treatment_report_functions.sql (hunk 2 above)
Comment (lauravignoli) [reply reportssql/treatment_report_functions.sql:464]: Thanks, will fix it.

File: reportssql/treatment_report_functions.sql (hunk 1 above)
Comment (tmvumbi2) [reply reportssql/treatment_report_functions.sql]: @lauravignoli sorry my comment wasn't clear enough, it was referring to the duration of the  treatment (i.e. not the interval provided as parameters)

**Pull request merged**